core_impl/src/main/java/com/google/instrumentation/trace/SpanFactoryImpl.java

```diff
@@ -0,0 +1,131 @@
+import java.util.EnumSet;
+import java.util.List;
+import java.util.Random;
+import javax.annotation.Nullable;
+
+/** Implementation of the {@link SpanFactory}. */
+final class SpanFactoryImpl extends SpanFactory {
+  private static final EnumSet<Span.Options> noRecordEventOptions =
+      EnumSet.noneOf(Span.Options.class);
+  private static final EnumSet<Span.Options> recordEventOptions = EnumSet.of(Options.RECORD_EVENTS);
+
+  private final RandomHandler randomHandler;
+  private final SpanImpl.StartEndHandler startEndHandler;
+  private final Clock clock;
+  private final TraceConfig traceConfig;
+
+  @Override
+  protected Span startSpan(@Nullable Span parent, String name, StartSpanOptions options) {
+    SpanContext parentContext = null;
+    TimestampConverter timestampConverter = null;
+    if (parent != null) {
+      parentContext = parent.getContext();
+      // Pass the timestamp converter from the parent to ensure that the recorded events are in
+      // the right order. Implementation uses System.nanoTime() which is monotonically increasing.
+      if (parent instanceof SpanImpl) {
+        timestampConverter = ((SpanImpl) parent).getTimestampConverter();
+      }
+    }
+    return startSpanInternal(parentContext, false, name, options, timestampConverter);
+  }
+
+  @Override
+  protected Span startSpanWithRemoteParent(
+      @Nullable SpanContext remoteParent, String name, StartSpanOptions options) {
+    return startSpanInternal(remoteParent, true, name, options, null);
+  }
+
+  Span startSpanInternal(
+      @Nullable SpanContext parent,
+      boolean hasRemoteParent,
+      String name,
+      StartSpanOptions startSpanOptions,
+      TimestampConverter timestampConverter) {
+    TraceParams activeTraceParams = traceConfig.getActiveTraceParams();
+    Random random = randomHandler.current();
+    TraceId traceId;
+    SpanId spanId = SpanId.generateRandomId(random);
+    SpanId parentSpanId = null;
+    TraceOptions.Builder traceOptionsBuilder;
+    if (parent == null || !parent.isValid()) {
+      // New root span.
+      traceId = TraceId.generateRandomId(random);
+      traceOptionsBuilder = TraceOptions.builder();
+      // This is a root span so no remote or local parent.
+      hasRemoteParent = false;
+    } else {
+      // New child span.
+      traceId = parent.getTraceId();
+      parentSpanId = parent.getSpanId();
+      traceOptionsBuilder = TraceOptions.builder(parent.getTraceOptions());
+    }
+    Sampler sampler =
+        startSpanOptions.getSampler() != null
+            ? startSpanOptions.getSampler()
+            : activeTraceParams.getSampler();
+    if (sampler.shouldSample(
+        parent, hasRemoteParent, traceId, spanId, name, startSpanOptions.getParentLinks())) {
+      traceOptionsBuilder.setIsSampled();
+    }
+    TraceOptions traceOptions = traceOptionsBuilder.build();
+    EnumSet<Span.Options> spanOptions = noRecordEventOptions;
```

Comment:
Change this to:

EnumSet<Span.Options> spanOptions = EnumSet.noneOf(Span.Options.class);

To make it clear that there are no options set (doesn't assume that record events is the only option)